libs/wingc/src/type_check.rs

```diff
@@ -1740,25 +1823,48 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
-	fn add_module_to_env(&mut self, env: &mut SymbolEnv, module_name: String, statement_idx: usize) {
-		// TODO Hack: treat "cloud" or "std" as "_ in wingsdk" until I figure out the path issue
-		if module_name == "cloud" || module_name == "fs" || module_name == WINGSDK_STD_MODULE {
+	fn add_module_to_env(
+		&mut self,
+		env: &mut SymbolEnv,
+		library_name: String,
+		namespace_filter: Vec<String>,
+		identifier: &Symbol,
+		statement_idx: usize,
+	) {
+		let (wingii_types, assembly_name) = if library_name == WINGSDK_ASSEMBLY_NAME {
+			// in runtime, if "WINGSDK_MANIFEST_ROOT" env var is set, read it. otherwise set to "../wingsdk" for dev
+			let manifest_root = std::env::var("WINGSDK_MANIFEST_ROOT").unwrap_or_else(|_| "../wingsdk".to_string());
 			let mut wingii_types = wingii::type_system::TypeSystem::new();
 			let wingii_loader_options = wingii::type_system::AssemblyLoadOptions {
 				root: true,
 				deps: false,
 			};
-			// in runtime, if "WINGSDK_MANIFEST_ROOT" env var is set, read it. otherwise set to "../wingsdk" for dev
-			let wingsdk_manifest_root = std::env::var("WINGSDK_MANIFEST_ROOT").unwrap_or_else(|_| "../wingsdk".to_string());
-			let name = wingii_types
-				.load(wingsdk_manifest_root.as_str(), Some(wingii_loader_options))
+			let assembly_name = wingii_types
+				.load(manifest_root.as_str(), Some(wingii_loader_options))
+				.unwrap();
+			(wingii_types, assembly_name)
+		} else {
+			let mut wingii_types = wingii::type_system::TypeSystem::new();
+			let wingii_loader_options = wingii::type_system::AssemblyLoadOptions { root: true, deps: true };
+			let source_dir = self.source_path.parent().unwrap().to_str().unwrap();
+			let assembly_name = wingii_types
+				.load_dep(library_name.as_str(), source_dir, &wingii_loader_options)
```

Comment:
Added a comment. TL;DR it's because the SDK we are loading from a specific location (provided by the WINGSDK_MANIFEST_ROOT environment variable) whereas other modules we need to search for from the user's project root directory. One is doing a lookup and one is not, hence the different arguments.

There might be better ways to clean this up but I can't quite envision what the better API would be -- definitely something we can iterate on